src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -698,9 +698,68 @@ export default class QuestionnaireForm extends Component {
     return c;
   }
 
+  addAuthorToResponse(qr, practionerRef)
+  {
+    function traverseToItemsLeafNode(item, url)
+    {
+      if (!item.item) {
+          return addAuthor(item,url);
+      }
+      else {
+          item.item.map(item =>
+              {
+                traverseToItemsLeafNode(item, url);
+              })
+              
+          }
+        
+    }
+    // url is a string
+    function addAuthor(item, practionerRef)
+    {
+      if (item.extension)
+      {
+          item.extension.push(
+              {
+                  "url": "http://hl7.org/fhir/StructureDefinition/questionnaireresponse-author",
+                  "valueReference": 
+                  {
+                      "reference": practionerRef
+                  }
+              }
+          )
```

Comment:
You can define an object before check wether the item has extension or not, then use the object in both cases to reduce duplicate code.